### PR TITLE
[16.0][IMP] shopinvader_product: main variant: add hooks

### PR DESCRIPTION
**Pull request by marielejeune**
Forward-port of https://github.com/shopinvader/odoo-shopinvader/pull/1529

### CODE REVIEW

**Comment by simahawk** [comment]
@marielejeune can you please review this https://github.com/shopinvader/odoo-shopinvader/pull/1529 and fwd port here instead?

File: shopinvader_product/models/product_product.py [hunk 1]
```diff
@@ -64,34 +64,51 @@ def _get_shopinvader_product_variants(self, product_ids):
...
+    def _get_main_product_read_fields(self):
+        product_model = self.env["product.product"]
+        order_by = [x.strip() for x in product_model._order.split(",")]
+        return ["product_tmpl_id"] + [f.split(" ")[0] for f in order_by]
+
+    @api.model
+    def _get_main_product_sorted_variants(self, variants):
+        # NOTE: if the order is changed by adding `asc/desc` this can be broken
+        # but it's very unlikely that the default order for product.product
+        # will be changed.
```
Comment (qgroulard) [inline shopinvader_product/models/product_product.py]: This note can be removed, it is not true anymore.

File: shopinvader_product/models/product_product.py [hunk 2]
```diff
@@ -64,34 +64,51 @@ def _get_shopinvader_product_variants(self, product_ids):
...
+        order_by = [x.strip() for x in self.env["product.product"]._order.split(",")]
```
Comment (qgroulard) [inline shopinvader_product/models/product_product.py:75]: `order_by` used to be the same as the one used to compute `fields_to_read`. I'm a bit lost, but shouldn't we keep this consistency ? @simahawk ?

Comment (marielejeune) [reply shopinvader_product/models/product_product.py:75]: You would like to extract the compute of `order_by` from `_get_main_product_read_fields` into a new hook to re-use it here into `_get_main_product_sorted_variants` right?

File: shopinvader_product/models/product_product.py (hunk 1 above)
Comment (marielejeune) [reply shopinvader_product/models/product_product.py]: done

File: shopinvader_product/models/product_product.py (hunk 2 above)
Comment (qgroulard) [reply shopinvader_product/models/product_product.py:75]: If necessary, yes.

**Comment by marielejeune** [comment]
It seems that there is a problem with Codecov but I cannot rerun the job.

**Review by lmignon — APPROVED** [review]

**Pull request merged**